classes/controlledVocab/ControlledVocab.php

```diff
@@ -1,108 +1,199 @@
  *
  * @class ControlledVocab
  *
- * @ingroup controlled_vocab
- *
- * @see ControlledVocabDAO
- *
- * @brief Basic class describing an controlled vocab.
+ * @brief ControlledVocab model class
  */
 
 namespace PKP\controlledVocab;
 
-use PKP\db\DAORegistry;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+use PKP\controlledVocab\ControlledVocabEntry;
+use PKP\facades\Locale;
 
-class ControlledVocab extends \PKP\core\DataObject
+class ControlledVocab extends Model
 {
-    //
-    // Get/set methods
-    //
+    use HasCamelCasing;
 
     /**
-     * get assoc id
-     *
-     * @return int
+     * List of pre defined vocab symbolic as const in format of CONTROLLED_VOCAB_*
+     */
+    public const CONTROLLED_VOCAB_SUBMISSION_AGENCY = 'submissionAgency';
+    public const CONTROLLED_VOCAB_SUBMISSION_DISCIPLINE = 'submissionDiscipline';
+    public const CONTROLLED_VOCAB_SUBMISSION_KEYWORD = 'submissionKeyword';
+    public const CONTROLLED_VOCAB_SUBMISSION_LANGUAGE = 'submissionLanguage';
+    public const CONTROLLED_VOCAB_SUBMISSION_SUBJECT = 'submissionSubject';
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Model::$table
+     */
+    protected $table = 'controlled_vocabs';
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Model::$primaryKey
+     */
+    protected $primaryKey = 'controlled_vocab_id';
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\GuardsAttributes::$guarded
+     */
+    protected $guarded = [
+        'controlled_vocab_id',
+    ];
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\HasTimestamps::$timestamps
+     */
+    public $timestamps = false;
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\HasAttributes::casts
      */
-    public function getAssocId()
+    protected function casts(): array
     {
-        return $this->getData('assocId');
+        return [
+            'symbolic' => 'string',
+            'assoc_type' => 'integer',
+            'assoc_id' => 'integer',
+        ];
     }
 
     /**
-     * set assoc id
-     *
-     * @param int $assocId
+     * Accessor and Mutator for primary key => id
      */
-    public function setAssocId($assocId)
+    protected function id(): Attribute
     {
-        $this->setData('assocId', $assocId);
+        return Attribute::make(
+            get: fn ($value, $attributes) => $attributes[$this->primaryKey] ?? null,
+            set: fn ($value) => [$this->primaryKey => $value],
+        )->shouldCache();
     }
 
     /**
-     * Get associated type.
-     *
-     * @return int
+     * Get the list of pre defined vocab symbolics
      */
-    public function getAssocType()
+    public static function getDefinedVocabSymbolic(): array
```

Comment:
Once you've cleaned up the symbolic name vs. setting name confusion, I think you can get rid of this function. I'd like to avoid hard-coding a list of all "acceptable" controlled vocabs, because third party code (e.g. plugins) might well want to extend the list.

Reply:
@asmecher I can get rid of this function however in that case, we need to define each of these const for the settings and multilingual definition for `non schema based entity` , more details at https://github.com/pkp/pkp-lib/pull/10324#discussion_r1888497825 .